java/arcs/core/data/Schema.kt

```diff
@@ -37,6 +37,12 @@ data class Schema(
     val refinementExpression: Expression<Boolean> = true.asExpr(),
     val queryExpression: Expression<Boolean> = true.asExpr()
 ) {
+
+    /** Ensure instance is registered on construction. */
+    init {
+        SchemaRegistry.register(this)
```

Comment:
this is already in the generated schemas: https://source.corp.google.com/piper///depot/google3/third_party/javascript/arcs/src/tools/kotlin-entity-generator.ts;rcl=322913420;l=242
which will be invoked the first time that schema is referenced anywhere.
So I doubt this will fix the problem.

Reply:
There's a small difference from this PR and the register inside code-generation. 
Code-gen registration happens when an Entity is reference (i.e. it's companion object is loaded). This PR will register schemas anytime they are constructed. 

In any case, this PR should only help and won't hurt.

Reply:
as discussed in chat we are sometimes creating schemas without using the code generated thing. so this fix should work!